pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -110,69 +52,177 @@ def check_parser_compatibility(cls, file):
         contains_engine = "Comet" in head
         return is_mzid and contains_engine
 
-    def _map_mods_and_sequences(self):
-        """Replace internal tags to retrieve sequences and formatted modification strings.
+    def get_version(self):
+        """Retrieve version from xml.
 
-        Operations are performed inplace.
+        Returns:
+            version (str): file version
```

Comment:
function does not return version as str but None if `elif entry_tag.endswith("AnalysisSoftware"):` is not True

Reply:
I'd also say dont put return in conditionals but set a value in the conditional and return it in the end.
You can always set None as default in the beginning if this is desired behavior

Reply:
I've now set it up exactly as you described.
Returning None is actually never desired - should always return the version, nothing else...